code/common/subscriptions.q

```diff
@@ -69,16 +69,17 @@ replay0:{[tabs;realsubs;schemalist;logfilelist;filters]
   if[not (tabs;realsubs[`instrs])~(`;`);
     .lg.o[`subscribe;"using the .sub.replayupd function as not replaying all tables or instruments"];
     @[`.;`upd;:;.sub.replayupd[origupd;subtabs;realsubs[`instrs]]]];
-  // replays log files and applies filters if in datastriping mode
-  f:{[lf;td]
-  // lf is a log file handle and td is a dictionary with table names as keys and where clauses to filter by as values
-    .lg.o[`subscribe;"replaying log file ",.Q.s1 lf]; -11!lf;
-    if[.ds.datastripe; .[set;] each (key td),'enlist each replayfilter each (key td),' value enlist each td;  
-       .lg.o[`subscribe;"filtering tables ", .Q.s1 key td]]
-    };
-  {[d;filter;func] .[func;(d;filter);{.lg.e[`subscribe;"could not replay the log file: ", x]}]}[;filters;f] each logfilelist;
-
-
+  // if datastriping is on replays only logs from current periods and applys filtering
+  if[.ds.datastripe;
+    //get tplog metadata table from stp  
+    logmetatab:@[proc`w;({@[value;`.stpm.metatable;`]};`);`];
```

Comment:
if this fails and gets set to `` ` `` - won't the `exec` on the next line fail anyway?

Reply:
Oh yeah, should I remove the error trap then?

Reply:
yeah I think might as well just fail here - I'm assuming if we don't get the metatable for some reason, we can't continue?

Reply:
Yeah it won't be able to continue without the metatable

Reply:
cool, yeah. I think we should just "fail fast" then

Reply:
Okay can make that change now, thanks